src/main/kotlin/AppModule.kt

```diff
@@ -47,49 +50,65 @@ fun Application.module() {
         }
         route("/") {
             get {
-                call.respond(ThymeleafContent("index", mapOf("obj" to "")))
+                call.respond(ThymeleafContent("index", mapOf("obj" to ""), locale = Locale("vi")))
             }
-            post {
-                val data = call.receive<FilePathTestModel>()
-                val serializedResult = processFileUpload(data)
+        }
+        post("/upload") {
+            val multiPart = call.receiveMultipart()
+            val dataMap: MutableMap<String, String> = mutableMapOf()
+            val fileUploadModel: FileUploadModel
+            var file = File("")
 
-                call.respondText(serializedResult, ContentType.Application.Json)
+            multiPart.forEachPart { part ->
+                when (part) {
+                    is PartData.FormItem -> {
+                        dataMap[part.name!!] = part.value
+                    }
+                    is PartData.FileItem -> {
+                        file = File("./src/main/resources/uploads/${part.originalFileName!!}")
+                        part.streamProvider().use { input ->
+                            file.outputStream().buffered().use {
+                                input.copyTo(it)
+                            }
+                        }
+                    }
+                }
+                part.dispose()
             }
-        }
-        get("/greeting") {
-            call.respond(ThymeleafContent("greeting_page", mapOf("user" to ""), locale = Locale.FRENCH))
+
+            val result = processFileUpload(file, dataMap)
+
+            call.respondText(result)
         }
     }
 }
 
-private fun processFileUpload(data: FilePathTestModel): String {
-    val model = if (data.mediaQuality.isNotBlank()) {
-        FilePathTestModel(
-            fileName = data.fileName,
-            languageCode = data.languageCode,
-            dublinCoreId = data.dublinCoreId,
-            projectId = data.projectId,
-            grouping = data.grouping,
-            mediaExtension = data.mediaExtension,
-            mediaQuality = data.mediaQuality,
-            expectedResult = ""
+private fun processFileUpload(file: File, data: Map<String,String>): String {
```

Comment:
I think that this function may need to be in another file. It seems clunky to have the functionality to handle the process in the same place that the route for the post request is being handled. I would consider only receiving the data with the call.receiveMultipart() function and then taking that and feeding it into another class. That class could then have methods to process the file upload and give a response.